packages/react-dom/src/__tests__/ReactDOMComponent-test.js

```diff
@@ -2532,6 +2532,45 @@ describe('ReactDOMComponent', () => {
 
       expect(el.getAttribute('hidden')).toBe('');
     });
+
+    it('warns on the ambiguous string value "" when it means false', function() {
+      let el;
+      expect(() => {
+        el = ReactTestUtils.renderIntoDocument(<div hidden="" />);
+      }).toWarnDev(
+        'Received the string "" for the boolean attribute `hidden`. ' +
+          'This value can mean `true` or `false`, depending on the attribute. ' +
+          'Did you mean hidden={false}?',
+      );
+
+      expect(el.hasAttribute('hidden')).toBe(false);
+    });
+
+    it('warns on the ambiguous string value "" when it means true', function() {
```

Comment:
Arguably this test belongs under `String boolean attributes` above, rather than here under `Boolean attributes` (a test category I introduced in #13372), but then again it's clearly a sibling test to the other two that do belong here. 🤷‍♂️  Happy to restructure this based on feedback.